src/Mvc/test/Mvc.FunctionalTests/ReadFromDisconnectedClientTest.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests
+{
+    // These tests verify the behavior of MVC when responding to a client that simulates a disconnect.
+    // See https://github.com/dotnet/aspnetcore/issues/13333
+    public class ReadFromDisconnectedClientTest : IClassFixture<MvcTestFixture<BasicWebSite.StartupWhereReadingRequestBodyThrows>>
```

Comment:
Like the functional tests. Would appreciate new unit tests of `FormValueProviderFactory` and `JQueryFormValueProviderFactory`. May also need additional confirmation of the `InvalidDataException` cases.

Reply:
The antiforgery one was hard to verify without an integration test, the rest are fairly easy to unit test so I added some coverage for those.